File: 2019/1/prob01a.java
import java.io.*;

public class prob01a {
  public int fuel(int m) {
    return m / 3 - 2;
  }

  public static void main(String[] args) {
    try {
      FileReader in = new FileReader("input.txt");

      int sum = 0;
      String buffer = "";
      int c;
      while (true) {
        c = in.read();
        if (((char) c) == '\n' || c == -1) {
          sum += fuel(Integer.parseInt(buffer.trim()));
          buffer = "";
        } else {
          buffer += (char) c;
        }

        if (c == -1)
          break;
      }

      System.out.println(sum);
      in.close();
    } catch (IOException fe) {
      System.out.println("File not found!");
    }
  }
}

File: 2019/1/prob01a.exs
defmodule Prob01a do
  def sum(file) do
    file
      |> Enum.map(&Integer.parse/1)
      |> Enum.map(&fuel/1)
      |> Enum.sum
  end

  def fuel({m, _}) do
    m / 3 - 2
      |> Kernel.trunc
  end
end

File.stream!("input.txt", [:read])
  |> Prob01a.sum
  |> IO.puts
